ReSharper.FSharp/src/FSharp.TypeProvidersProtocol/src/Utils/ProvidedTypesComparer.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
+using static FSharp.Compiler.ExtensionTyping;
+
+namespace JetBrains.ReSharper.Plugins.FSharp.TypeProvidersProtocol.Utils
+{
+  public class ProvidedTypesComparer : IEqualityComparer<ProvidedType>
+  {
+    private ProvidedTypesComparer()
+    {
+    }
+    //TODO: check and comment
```

Comment:
Yep. I am almost convinced this implementation is good enough for our purposes, but please add a comment.